### ISSUE TITLE
How to specify the grouping function for bullet points?

### ISSUE THREAD

**Issue by nothing-personal-gh** [amcharts/amcharts5#563]
Hi,

I am trying to use '**average**' instead of the default '**last**' grouping function for bullets shown in an XY chart.

I have set 'sum' for `valueYGrouped` and it sums the Y values perfectly. I also have a bullet for each point along with the data values. I need to use **average** when multiple bullets are grouped as per the `groupIntervals`. But amcharts 5 seems to use the 'last' as the default aggregation type. Where do I specify that I want to use 'average' for bullets?

Checkout this sample chart - https://codepen.io/trooper-isa/pen/dymrXXB?editors=0010

Hover on the first bullet. The bullet shows **60 degrees** for the first data point (because **60** is the last value in the group) but I want to show the average.

amCharts version: 5
Browser and its version: Vivaldi - 5.4.2753.37 (Stable channel) (x86_64)

**Comment by martynasma** [comment]
OK, so there are a couple of steps to it:

1) Use custom grouping function to calculate averages for each group data point and set it to data item's `valueYAverage`:

```JavaScript
var series = chart.series.push(
  am5xy.LineSeries.new(root, {
    name: "Series",
    xAxis: xAxis,
    yAxis: yAxis,
    valueYField: "value",
    valueXField: "time",
    valueYGrouped: "sum",
    tooltip: am5.Tooltip.new(root, {
      labelText: "{valueY}"
    }),
    groupDataCallback: function(dataItem, interval) {
      var originals = dataItem.get("originals");
      var avg = 0;
      am5.array.each(originals, function(original) {
        avg += original.get("valueY");
      });
      avg /= originals.length;
      dataItem.set("valueYAverage", avg)
    }
  })
);
```

2) Use `valueYAverage` in tooltip:

```JavaScript
tooltip.label.setAll({
  text: `{valueYAverage.formatNumber('${numberFormat}')} degrees`
});
```

Updated CodePen:
https://codepen.io/team/amcharts/pen/WNzmRXE/19e8b71592a3883216d91d86562cf0c4?editors=0010

**Comment by nothing-personal-gh** [comment]
@martynasma thanks for the reply.

I would like to keep the original tooltip with the 'sum' value of `value` field. This is set on `valueYGrouped` property already and working correctly. Eg. 6 for the first data point. 

I need to do 'average' only for the bullet field. I have added the `groupDataCallback` now.

```
    groupDataCallback: function (dataItem, interval) {
      var originals = dataItem.get("originals");
      var avg = 0;
      am5.array.each(originals, function (original) {
        avg += original.get("bullet");
      });
      avg /= originals.length;
      dataItem.set("bulletAverage", avg);
    },
```

I have used `bulletAverage` for the field name instead of `valueYGrouped` as this is not related to the `value` field. 

However, this field is not available in:

```
  const bulletDegree = _.get(
    target,
    "dataItem.dataContext.bulletAverage",
    undefined
  );
```

I get undefined.

<img width="291" alt="image" src="https://user-images.githubusercontent.com/111722528/186024531-229362b9-ebc6-41c7-a1c1-9ccd03bb3b40.png">

Am I reading it correctly?

Updated code in the same codepen - https://codepen.io/trooper-isa/pen/dymrXXB?editors=0010
